frontend/mainwindow.h

```diff
@@ -27,18 +28,26 @@ class MainWindow : public QMainWindow {
   void initialiseElements();
 private slots:
   void on_filterBox_textChanged(const QString &arg1);
+  void on_searchBox_textChanged(const QString &arg1);
   void handle_result_instruments(HttpRequestWorker *worker);
   void handle_result_cycles(HttpRequestWorker *worker);
-  void on_instrumentsBox_currentTextChanged(const QString &arg1);
+  void instrumentsBoxChange(const QString &arg1);
   void on_cyclesBox_currentTextChanged(const QString &arg1);
   void on_groupButton_clicked(bool checked);
   void columnHider(int state);
+  void on_clearSearchButton_clicked();
+  void on_findUp_clicked();
+  void on_findDown_clicked();
+  void on_searchAll_clicked();
+  void recentCycle();
 
 private:
   Ui::MainWindow *ui;
   JsonTableModel *model;
   QSortFilterProxyModel *proxyModel;
   QMenu *viewMenu;
   JsonTableModel::Header header;
+  std::tuple<QModelIndexList, int> matchesTuple;
```

Comment:
Not sure having this data paired in a `std::tuple` adds any value - it makes the code harder to understand for sure!  Separate member variables (e.g. `foundIndices_` and `currentFoundIndex_` might be a better way to go.

Reply:
I thought it would be nicer to have the data together but definitely agree that it's probably too messy